src/imex.rs

```diff
@@ -706,6 +711,19 @@ async fn export_database(
         .await
 }
 
+/// Sets `Config::DeleteServerAfter` to "never" if needed so that new messages are present on the
+/// server after a backup restoration or available for all devices in multi-device case.
+/// NB: Calling this after a backup import isn't reliable as we can crash in between, but this is a
+/// problem only for old backups, new backups already have `DeleteServerAfter` set if necessary.
+async fn adjust_delete_server_after(context: &Context) -> Result<()> {
+    if context.config_exists(Config::DeleteServerAfter).await? || !context.is_chatmail().await? {
+        return Ok(());
+    }
+    context
+        .set_config(Config::DeleteServerAfter, Some("0"))
+        .await
```

Comment:
Nit, feel free to ignore: I think this code would be easier to read like this:
```suggestion
    if context.is_chatmail().await? && !context.config_exists(Config::DeleteServerAfter).await? {
        context
            .set_config(Config::DeleteServerAfter, Some("0"))
            .await?;
    }
    Ok(())
```
because it spells out more directly what the code should do ("For chatmail servers, if the config is still the default, set it to 'never'")